balancer/balancer-subgraph-v2
# Dev

# Description

- #445 
- #442 
- #443 
- #447 
- #449 
- #448 
- #452 
- #430 
- #451 

## Type of change


## How should this be tested?

- Linear pools that are in recovery mode show up as such ([see euler linear pools](https://api.thegraph.com/subgraphs/name/balancer-labs/balancer-v2-beta/graphql?query=%7B%0A++pools%28where%3A%7B%0A++++poolType_contains%3A%22Linear%22%0A++++isInRecoveryMode%3Atrue%0A++%7D%0A++%29+%7B%0A++++id%0A++%7D%0A%7D))

- Vault's share of pools are reasonable amounts ([check](https://api.thegraph.com/subgraphs/name/balancer-labs/balancer-v2-beta/graphql?query=%7B%0A++poolShares+%28where%3A%7B%0A++++userAddress%3A%220xba12222222228d8ba445958a75a0704d566bf2c8%22%0A++%7D%0A++++orderBy%3Abalance%0A++++orderDirection%3Adesc%0A%29+%7B%0A++++id%0A++balance%0A++%7D%0A%7D))

- Internal balance of BPT counts to user's pool shares ([check](https://api.thegraph.com/subgraphs/name/balancer-labs/balancer-v2-beta/graphql?query=%7B%0A++userInternalBalances+%28block%3A%7Bnumber%3A17293252%7D%0A++++where%3A%7B%0A++++token%3A%220xa13a9247ea42d743238089903570127dda72fe44%22%0A++++userAddress%3A%220x000000000dfde7deaf24138722987c9a6991e2d4%22%0A++%7D%29%7B%0A++++balance%0A++%7D%0A++poolShares+%28block%3A%7Bnumber%3A17293252%7D+where%3A%7B%0A++++userAddress%3A%220x000000000dfde7deaf24138722987c9a6991e2d4%22%0A++++poolId%3A%220xa13a9247ea42d743238089903570127dda72fe4400000000000000000000035d%22%0A++%7D%29+%7B%0A++++id%0A++++balance%0A++%7D%0A%7D))

## Checklist:


### `dev` -> `master`
- [x] I have [checked](https://balancer.github.io/balancer-subgraph-v2/status.html) that all beta deployments have synced
- [x] I have [checked](https://api.thegraph.com/subgraphs/name/balancer-labs/balancer-polygon-prune-v2-beta/graphql?query=%0A%7B%0A++balancers%28block%3A%7Bnumber%3A1%7D%29%7B%0A++++id%0A++%7D%0A%7D) that the earliest block in the polygon pruned deployment is [42745087, May-15-2023](https://polygonscan.com/block/42745087)
  - [x] The earliest block is more than 24 hours old
- [x] I have checked that core metrics are the same in the beta and production deployments
